### DISCUSSION TITLE
Using the asistant !

### DISCUSSION THREAD

**Discussion by enzime4u** [zed-industries/zed#6650] [Feature Requests]
Hi there, 
I have a paid Github Copillot and a free OpenAI account. It seems that the Assistant is not working at all , I tried to copy code and format is etc etc .... I am missing something or what ? 
Thank you in advance....

**Comment by JosephTLyons** [comment]
Hey @enzime4u, the assistant uses [OpenAI's API](https://openai.com/pricing), which is a separate service that isn't free, unfortunately. It is confusing because OpenAI allows you to create API keys, even if you aren't paying for the API service.

**Reply by enzime4u** [reply]
yes I do have an OPEN AI's key entered , this means that I should pay for OpenAI too ? or for Github copilot ?
Thank you

**Reply by JosephTLyons** [reply]
To use the assistant, you have to pay for OpenAI's API. In the future, we will likely add support for other AI services, including free services, so that this is more accessible.

**Reply by ralgozino** [reply]
I faced the same issue today, I configured my (free) OpenAI key and the assistant wasn't working.

I haven't checked, but if there's a way to check if the API key is working and has the right permission (i.e. not a free account) I'd like to suggest improving the UX by showing the user a message that the provided API key is not good. This would mean hard-coding something that OpenAI could change in the future, if you prefer to not do that, I would say that adding a warning in the [assistant section of the docs](https://docs.zed.dev/general/assistant-panel#setup) about the free account not working would also be appreciated

thanks!

**Reply by SMak06** [reply]
How do I change/remove a key once I entered it?

**Reply by Moshyfawn** [reply]
> How do I change/remove a key once I entered it?

#8200
